runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java

```diff
@@ -202,7 +202,15 @@ public void run(SourceContext<WindowedValue<ValueWithRecordId<OutputT>>> ctx) th
     ReaderInvocationUtil<OutputT, UnboundedSource.UnboundedReader<OutputT>> readerInvoker =
         new ReaderInvocationUtil<>(stepName, serializedOptions.get(), metricContainer);
 
-    if (localReaders.size() == 1) {
+    if (localReaders.size() == 0) {
+      // It can happen when value of parallelism is greater than number of IO readers (for example,
+      // parallelism is 2 and number of Kafka topic partitions is 1). In this case, we just idle
+      // this executor.
+      LOG.warn("Number of readers is 0, idle");
```

Comment:
Probably shouldn't even be a warning, since it's a completely normal thing that can just happen